### ISSUE TITLE
Freeze when finishing a play while deleting all beatmaps

### ISSUE THREAD

**Issue by LittleEndu** [ppy/osu#6334]
**Describe the crash:** When deleting all beatmaps and then going to play, when you finish the play the results screen (SoloResults) never appears and the game freezes 

**Screenshots or videos showing encountered issue:** Black screen with elements that remain after finishing the play, when the song finishes. I wasn't able to screenshot this, for some reason it shows main menu instead or is just black screen: https://i.imgur.com/yIcD2Jc.png

**osu!lazer version:** 2019.930.0

**Logs:** Nothing meaningful seemed to be logged but here they are: [logs.zip](https://github.com/ppy/osu/files/3672447/logs.zip)

Only thing interesting might be this database line: ``2019-09-30 05:19:07: [?????] Beginning import...``

**Computer Specifications:** Win10, Pentium 4405U, no graphics card

**Comment by smoogipoo** [comment]
Does it recover after the deletion completes?

**Comment by LittleEndu** [comment]
Task manager reported that the game eventually stopped responding. It might have been when the deletion finished so it definitely didn't seem to recover.

**Comment by LittleEndu** [comment]
While in windowed mode I got a screenshot: ![saidscreenshot](https://i.imgur.com/BhklDHs.png)

The crash happens only when there are still beatmaps to be deleted when finishing the play. I tried a medium length map with 100 beatmaps and that finished correctly, then I tried 1k beatmaps with 30 second song and that freezes 100% of the time.

This is now happening on a fresh install.

I also waited out this time, the fresh install did manage to recover after several minutes and never stopped responding.
